pkg/helm/chartproxy/proxy_test.go

```diff
@@ -16,16 +17,29 @@ import (
 
 func TestProxy_IndexFile(t *testing.T) {
 	tests := []struct {
-		name       string
-		indexFiles []string
-		mergedFile string
-		helmCRS    []*unstructured.Unstructured
+		name           string
+		indexFiles     []string
+		mergedFile     string
+		kubeVersion    *chartutil.KubeVersion
+		helmCRS        []*unstructured.Unstructured
+		onlyCompatible bool
 	}{
 		{
 			name:       "returned index file for configured helm repo",
 			indexFiles: []string{"testdata/sampleRepoIndex.yaml"},
 			mergedFile: "testdata/mergedSampleRepoIndex2.yaml",
 		},
+		{
+			name:       "returned index file for configured helm repo for provided kubernetes cluster filter",
```

Comment:
how about: retuned index file for configured helm repo contains only charts compatible for given cluster?

Reply:
we should also have a test where we none of charts are compatible with given cluster, i.e. the returned index should be empty.